pkgs/development/python-modules/keystoneauth1/default.nix

```diff
@@ -0,0 +1,53 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, iso8601
+, os-service-types
+, pbr
+, requests
+, stevedore
+
+, lxml
+, oauthlib
+, pyyaml
+, requests-kerberos
+, testresources
+, testtools
+}:
+
+buildPythonPackage rec {
+  pname = "keystoneauth1";
+  version = "4.2.1";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "18c9khnk31i23x8mhvyrlh9gvhdphi3ac2p1f59d1wzg4z6bz5ll";
+  };
+
+  propagatedBuildInputs = [
+    iso8601
+    os-service-types
+    pbr
+    requests
+    stevedore
+  ];
+
+  checkInputs = [
+    oauthlib
+    lxml
+    requests-kerberos
+    pyyaml
+    testtools
+    testresources
+  ];
+
+  # Circular dependencies on stestr, see https://bugs.launchpad.net/oslo.config/+bug/1893978
+  doCheck = false;
```

Comment:
If tests are not available, then please use `pythonImportsCheck` to import the most important modules. This isn't as good as unit tests, but will usually give a good indication of run-time errors. Please see [pythonImportsCheck documentation](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pythonimportscheck) for more information.